docs/references/config/config.toml.md

```diff
@@ -0,0 +1,2055 @@
+|                     | absolute directory path                         |
+
+The default relative path translates to `$CMTHOME/config/priv_validator_key.json`. In case `$CMTHOME` is unset, it
+defaults to `$HOME/.cometbft/config/priv_validator_key.json`.
+
+
+### priv_validator_state_file
+Path to the JSON file containing the last sign state of a validator  (more details [here](./priv_validator_state.json.md)).
+```toml
+priv_validator_state_file = "data/priv_validator_state.json"
+```
+
+| Value type          | string                                          |
+|:--------------------|:------------------------------------------------|
+| **Possible values** | relative directory path, appended to `$CMTHOME` |
+|                     | absolute directory path                         |
+
+The default relative path translates to `$CMTHOME/data/priv_validator_state.json`. In case `$CMTHOME` is unset, it
+defaults to `$HOME/.cometbft/data/priv_validator_state.json`.
+
+### priv_validator_laddr
+TCP or UNIX socket listen address for CometBFT that allows external consensus signing processes to connect.
+```toml
+priv_validator_laddr = ""
+```
+
+| Value type          | string                                                     |
+|:--------------------|:-----------------------------------------------------------|
+| **Possible values** | TCP Stream socket (e.g. `"tcp://127.0.0.1:26665"`)         |
+|                     | Unix domain socket (e.g. `"unix:///var/run/privval.sock"`) |
+
+When consensus signing is outsourced from CometBFT (typically to a Hardware Security Module, like a
+[YubiHSM](https://www.yubico.com/product/yubihsm-2) device), this address is opened by CometBFT for incoming connections
+from the signing service.
+
+Make sure the port is available on the host machine and firewalls allow the signing service to connect to it.
+
+More information on a supported signing service can be found in the [TMKMS](https://github.com/iqlusioninc/tmkms)
+documentation.
+
+### node_key_file
+Path to the JSON file containing the private key to use for node authentication in the p2p protocol (more details [here](./node_key.json.md)).
+```toml
+node_key_file = "config/node_key.json"
+```
+
+| Value type          | string                                          |
+|:--------------------|:------------------------------------------------|
+| **Possible values** | relative directory path, appended to `$CMTHOME` |
+|                     | absolute directory path                         |
+
+The default relative path translates to `$CMTHOME/config/node_key.json`. In case `$CMTHOME` is unset, it defaults to
+`$HOME/.cometbft/config/node_key.json`.
+
+### abci
+The mechanism used to connect to the ABCI application.
+````toml
+abci = "socket"
+````
+
+| Value type          | string     |
+|:--------------------|:-----------|
+| **Possible values** | `"socket"` |
+|                     | `"grpc"`   |
```

Comment:
It can also have the empty string `""`, which denotes the application is (or should be) linked together with CometBFT and the local ABCI will be used (ABCI calls become function calls).